rtgui/icmpanel.cc

```diff
@@ -631,10 +1116,81 @@ void ICMPanel::write(ProcParams* pp, ParamsEdited* pedited)
     }
 
+    int aintentVal = aRendIntent->getSelected();
+
+    if (aintentVal >= 0 && aintentVal < RI__COUNT) {
+        pp->icm.aRendIntent  = static_cast<RenderingIntent>(aintentVal);
+    } else {
+        pp->icm.aRendIntent  = rtengine::RI_RELATIVE;
+    }
+
     if (wTRC->get_active_row_number() == 0) {
         pp->icm.workingTRC = "none";
     } else if (wTRC->get_active_row_number() == 1) {
         pp->icm.workingTRC = "Custom";
+    } else if (wTRC->get_active_row_number() == 2) {
+        pp->icm.workingTRC = "bt709";
+    } else if (wTRC->get_active_row_number() == 3) {
+        pp->icm.workingTRC = "srgb";
+    } else if (wTRC->get_active_row_number() == 4) {
+        pp->icm.workingTRC = "22";
+    } else if (wTRC->get_active_row_number() == 5) {
+        pp->icm.workingTRC = "18";
+    } else if (wTRC->get_active_row_number() == 6) {
+        pp->icm.workingTRC = "lin";
+    }
+
+
+    if (will->get_active_row_number() == 0) {
+        pp->icm.will = "def";
+    } else if (will->get_active_row_number() == 1) {
+        pp->icm.will = "D41";
+    } else if (will->get_active_row_number() == 2) {
+        pp->icm.will = "D50";
+    } else if (will->get_active_row_number() == 3) {
+        pp->icm.will = "D55";
+    } else if (will->get_active_row_number() == 4) {
+        pp->icm.will = "D60";
+    } else if (will->get_active_row_number() == 5) {
+        pp->icm.will = "D65";
+    } else if (will->get_active_row_number() == 6) {
+        pp->icm.will = "D80";
+    } else if (will->get_active_row_number() == 7) {
+        pp->icm.will = "D120";
+    } else if (will->get_active_row_number() == 8) {
+        pp->icm.will = "stda";
+    } else if (will->get_active_row_number() == 9) {
+        pp->icm.will = "2000";
+    } else if (will->get_active_row_number() == 10) {
+        pp->icm.will = "1500";
+    }
+
+    if (wprim->get_active_row_number() == 0) {
+        pp->icm.wprim = "def";
+    } else if (wprim->get_active_row_number() == 1) {
+        pp->icm.wprim = "srgb";
+    } else if (wprim->get_active_row_number() == 2) {
+        pp->icm.wprim = "adob";
+    } else if (wprim->get_active_row_number() == 3) {
+        pp->icm.wprim = "prop";
+    } else if (wprim->get_active_row_number() == 4) {
+        pp->icm.wprim = "rec";
+    } else if (wprim->get_active_row_number() == 5) {
+        pp->icm.wprim = "aces";
+    } else if (wprim->get_active_row_number() == 6) {
+        pp->icm.wprim = "wid";
+    } else if (wprim->get_active_row_number() == 7) {
+        pp->icm.wprim = "ac0";
+    } else if (wprim->get_active_row_number() == 8) {
+        pp->icm.wprim = "bru";
+    } else if (wprim->get_active_row_number() == 9) {
+        pp->icm.wprim = "bet";
+    } else if (wprim->get_active_row_number() == 10) {
+        pp->icm.wprim = "bst";
+    } else if (wprim->get_active_row_number() == 11) {
+        pp->icm.wprim = "cus";
+    } else if (wprim->get_active_row_number() == 12) {
+        pp->icm.wprim = "cusgr";
     }
```

Comment:
Here is another good example for which a switch statement would look a lot better. I'm not sure what the compiler does, but if it keeps all the `->get_active_row_number()` calls intact, this is pretty inefficient.

Reply:
I would prefer something like this:

```
constexpr std::string wprims[13] = {"def", "srgb", "adob", "prop", "rec", "aces", "wid", "ac0", "bru", "bet", "bst", "cus", "cusgr"};
pp->icm.wprim = wprims[rtengine::LIM(wprim->get_active_row_number(), 0, 12)];
```

Reply:
^ Same thing for `pp->icm.will` and `pp->icm.workingTRC` of course...

Reply:
@heckflosse 

Ingo
As you know, I am a poor developer...
I just tried your "recommandation" but I get error messages...can you make this change, thank you

Jacques

Reply:
@heckflosse Even better would be using an `std::array<std::string, 13>` and relying on `std::array<>::size()` for `LIM()`. Do you think `LIM()` is the right idea, or should there be a fixed default?

Reply:
@Floessie I'm working at this thing atm. Please hold ;-)